template/context.go

```diff
@@ -70,3 +78,175 @@ func (ctx *Context) Expand(s string) (string, error) {
 
 	return buf.String(), nil
 }
+
+// PayloadContext provides a context for expanding a config or secret payload.
+// NOTE: Be very careful adding any fields to this structure with types
+// that have methods defined on them. The template would be able to
+// invoke those methods.
+type PayloadContext struct {
+	Context
+	Secrets secretPayloadGetter
```

Comment:
Presumably, this particular context is only available for contexts after env variables have been evaluated?